shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -521,12 +523,17 @@ public void writeFields(DataOutput out) throws IOException {
             out.writeLong(writerId.getMostSignificantBits());
             out.writeLong(writerId.getLeastSignificantBits());
             out.writeLong(eventNumber);
+            out.writeLong(previousEventNumber);
         }
 
         public static WireCommand readFrom(DataInput in, int length) throws IOException {
             UUID writerId = new UUID(in.readLong(), in.readLong());
             long offset = in.readLong();
-            return new DataAppended(writerId, offset);
+            Long previousEventNumber = null;
+            if (length == 32) {
```

Comment:
We should bump up the version and check the version rather than size.

Reply:
@tkaitchuck your take on this?

Reply:
As the new feature is purely a safety check, and does not change how functionality works, I don't see a reason to bump the version.

Reply:
Perhaps we need to have a separate discussion about chaining the protocol version. My current interpretation is that we need to change the version every time we add a new message or change the fields of an existing message. Any change to the protocol should induce a protocol version change, independent of its impact feature-wise.

But, if you are saying that it is a bit of waste that we bump up the version because no one really cares at this point, then I agree, but I'd suggest that we simply remove the length check then. The length check that I have commented on is implicitly a check for protocol version, so I think either we bump up the protocol version and check the version or we remove the check altogether.

Reply:
@fpj I am fine with removing the length check. the assumption here being that we wont run into the (very unlikely) situation where someone has an older version of server deployed. with that change I could also revert the new field from `Long` to `long`

Reply:
i agree that - either 1. bump version or 2. no length check.
2 implies we dont make any attempts for being backwards compatible until we sense that we need to support older version at which point we start doing version bumps. not having to do it buys us simplicity in the code.

Reply:
Have removed the length check. And also switched from `Long` to `long`

Reply:
The whole point of the version number is so that the client and server can detect if they are not compatible with each other. So we should not break backward compatibility without a version number bump. 
 
We however don't need to break compatibility. The code as written is correct and backward compatible.

Reply:
I created a PDP for the broader discussion. https://github.com/pravega/pravega/wiki/PDP-21:-Protocol-revisioning
I think this change falls under the category I have labeled "Supplemental information"